Dockerfile

```diff
@@ -16,6 +16,19 @@ RUN npm install
 COPY . /usr/src/app
 RUN npm install
 RUN npm run build
+RUN apk --update add git jq
+RUN version="$(npm -j ls tslint | jq -r .dependencies.tslint.version)" && \
+  printf "Pulling docs from TSLint %s\n" "$version" && \
+  git clone --quiet --branch "$version" https://github.com/palantir/tslint.git && \
+  cd tslint && \
+  npm install && \
+  npm run compile && \
+  npm run docs && \
+  cd .. && \
+  mkdir -p ./lib/docs && \
+  cp --recursive tslint/docs/_data/rules.json ./lib/docs/rules.json && \
+  rm -rf tslint
+RUN apk del git jq
```

Comment:
Unless the `apk` install and remove commands are excuted within the same `RUN` invocation, it won't reduce overall image size since they'll apply to different docker layers, all included within the image.

I'd recommend extracting lines 21-30 into a separate `bin/generate-docs` script and collapse the three RUN invocations into one:

```dockerfile
RUN apk --update add git jq && \
  version="$(npm -j ls tslint | jq -r .dependencies.tslint.version)" && \
  bin/generate-docs "$version" && \
  apk --purge del git jq
```